CONTRIBUTING.md

```diff
@@ -1,51 +1,84 @@
 
-## Submitting Changes ##
+## Submitting a Pull Request
 
-If you'd like to make a change to linkerd, you should do the
-following:
+Do you have an improvement?
 
 1. Submit an [issue][issue] describing your proposed change.
-2. We will respond to your issue promptly.
-3. If your proposed change is accepted, and you haven't already done
-so, sign [Buoyant's Contributor License Agreement][cla].  Before we
-can accept your patches, our lawyers would like to be assured that:
-    - The code you're contributing is yours, and you have the right to
-    license it.
-    - You're granting us a license to distribute said code under the
-    terms of this agreement.
-4. Fork the linkerd repo, develop and test your code
-changes. See the project's [README](README.md) for further information
-about working in this repository.
-5. Submit a pull request against linkerd's `master` branch.
-6. Your branch may be merged once all configured checks pass,
-including:
+2. We will try to respond to your issue promptly.
+3. If your proposed change is accepted, and you haven't already done so, sign our [Contributor License Agreement][cla].
+4. Fork this repo, develop and test your code changes. See the project's [README](README.md) for further information about working in this repository.
+5. Submit a pull request against this repo's `master` branch.
+6. Your branch may be merged once all configured checks pass, including:
     - 2 code review approvals, at least 1 of which is from a buoyant member
     - The branch has passed tests in CI.
 
-### Git history ###
+## Code Style
 
-Prior to submitting a pull request, please revise the commit history
-of your branch such that each commit is self-explanatory, even without
-the context of the pull request, and that it compiles & tests
-cleanly.
+We generally follow [Effective Scala][es] and the
+[Scala Style Guide][ssg]. When in doubt, we try to use Finagle's
+idioms. 
 
-Typically, a pull request consists of a single commit.  It may
-occasionally be preferable to submit a change with several commits,
-but each should be a complete change needed to complete a larger
-feature.  Each commit should be documented appropriately.
+## Committing
 
-Thank you for getting involved!
-:heart: Team Buoyant
+We prefer squash or rebase commits so that all changes from a branch are committed to
+master a single commit.
+
+### Commit messages
+
+Finalized commit messages should be in the following format:
+
+```
+Subject
+
+Problem
+
+Solution
+
+Validation
+```
+
+#### Subject
+
+- one line, <= 50 characters
+- describe what is done; not the result
+- use the active voice
+- capitalize properly
+- do not end in a period — this is a title/subject
+- reference the pull request by number
+
+##### Examples
+
+bad: server disconnects should cause dst client disconnects.
+good: Propagate disconnects from source to destination
+
+bad: support tls servers
+good: Introduce support for server-side TLS (#347)
```

Comment:
These should be bulleted or separated by an additional line. Otherwise github combines them into one paragraph.

![screen shot 2017-04-06 at 11 31 59 am](https://cloud.githubusercontent.com/assets/9226/24769825/b11eda9a-1abc-11e7-9c16-aae3c72691ec.png)